components/nifosg/matrixtransform.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef OPENMW_COMPONENTS_NIFOSG_MATRIXTRANSFORM_H
+#define OPENMW_COMPONENTS_NIFOSG_MATRIXTRANSFORM_H
+
+#include <components/nif/niftypes.hpp>
+
+#include <osg/MatrixTransform>
+
+namespace NifOsg
+{
+
+    class MatrixTransform : public osg::MatrixTransform
```

Comment:
Why didn't you encapsulate the rotation and scale change into this type? Every time an object of this type is used a developer have to remember and duplicate the code to change underlying 4x4 matrix accoring to mScale and mRotationScale. You could say there is only one but it doesn't make this code less fragile.

Reply:
You mean the dancing around that happens in KeyframeController?

Reply:
Alright, should be done.